activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java

```diff
@@ -42,28 +47,61 @@ public class KeycloakManagementService implements IdentityManagementService {
     private final KeycloakUserToUser keycloakUserToUser;
     private final KeycloakGroupToGroup keycloakGroupToGroup;
     private final KeycloakTokenToUserRoles keycloakTokenToUserRoles;
+    private final KeycloakRoleMappingToRole keycloakRoleMappingToRole;
 
     public KeycloakManagementService(KeycloakClient keycloakClient,
         KeycloakUserToUser keycloakUserToUser,
         KeycloakGroupToGroup keycloakGroupToGroup,
-        KeycloakTokenToUserRoles keycloakTokenToUserRoles) {
+        KeycloakTokenToUserRoles keycloakTokenToUserRoles,
+        KeycloakRoleMappingToRole keycloakRoleMappingToRole) {
         this.keycloakClient = keycloakClient;
         this.keycloakUserToUser = keycloakUserToUser;
         this.keycloakGroupToGroup = keycloakGroupToGroup;
         this.keycloakTokenToUserRoles = keycloakTokenToUserRoles;
+        this.keycloakRoleMappingToRole = keycloakRoleMappingToRole;
     }
 
     @Override
     public List<User> findUsers(UserSearchParams userSearchParams) {
-        return keycloakClient
+        List<User> users= keycloakClient
             .searchUsers(userSearchParams.getSearchKey(), PAGE_START, PAGE_SIZE)
             .stream()
             .map(keycloakUserToUser::toUser)
+            .collect(Collectors.toList());
+
+        if(!StringUtils.isEmpty(userSearchParams.getApplication())) {
+            return filterUsersInApplicationsScope(users, userSearchParams);
+        } else {
+            return filterUsers(users, userSearchParams);
+        }
+    }
+
+    private List<User> filterUsers(List<User> users ,UserSearchParams userSearchParams) {
```

Comment:
just to double-check, the idea behind the `filterUsers` and `filterUsersInApplicationsScope` is that we provide both filtering options for single & multiple application?
in the same way we offer the possibility of setting `keycloak.use-resource-role-mappings` to both `false` and `true` ?

Reply:
If with "single application" you mean the realm level roles, yes.
In the basic scenario, we search for users and if we want to filter by their role we check the realm roles.
If we pass the parameter `application` then we return:

- the users that have access to the application (have at least one role in the Keycloack "client" role mapping)
- if we pass application and roles, then we return users that have that role in the client ignoring the REALM roles

Reply:
Understood, thanks for the explanation 👌

Reply:
Are we taking into account multiple apps ?

Reply:
No, the parameter `application` is a String, not a list.